### ISSUE TITLE
Compatibility with LAMMPS after 2Jun2022

### ISSUE THREAD

**Issue by xyttyxy** [mir-group/pair_nequip#41]
In https://github.com/lammps/lammps/pull/3199 a change was made to neighbor list request API, making full, half, and ghost not directly accessible to third-party code. This is used here https://github.com/mir-group/pair_nequip/blob/b3680ef1288747fd8694693c1a747be4dd6a76b9/pair_nequip.cpp#L97-L101

Hence pair_nequip will not compile with LAMMPS after 2Jun2022. A simple fix is by replacing the 3 lines cited by the new API:
```c++
neighbor->add_request(this, NeighConst::REQ_FULL);
```

**Comment by Linux-cpp-lisp** [comment]
Hi @xyttyxy ,

Thanks for your interest in our codes!

I think @anjohan fixed this in `pair_allegro`, but maybe the fix never made it here into `pair_nequip`. Thanks for letting us know!

**Comment by JunsuAndrewLee** [comment]
Hi @xyttyxy and @Linux-cpp-lisp ,

Thank you for your comments.

I was trying to integrate NequIP into LAMMPS 23Jun2022 version, 
With your modification cmake and make commands build the executable lmp without an error, however the executable does not recognize pair style nequip at the beginning of a LAMMPS run as follows. Would you let me know any further modifications I need to modify?

```
ERROR: Unrecognized pair style 'nequip' (src/force.cpp:279)
Last command: pair_style	nequip
```

**Comment by UMBC-STEAM-LAB** [comment]
Hi, 

I have the same problem for NequIP **(Unrecognized pair style 'nequip')** with LAMMPS 23Jun2022 version.

**Comment by Akram-Ibrahim** [comment]
Yeah, I've the same issue .. **Exception: ERROR: Unrecognized pair style 'nequip' (src/force.cpp:279)** .. with LAMMPS (23 Jun 2022 - Update 1).

**Comment by JunsuAndrewLee** [comment]
@Akram-Ibrahim Recently I managed to integrate nequip to LAMMPS 23Jun2022 stable version, though slightly different version from what you have tried. Did you make sure patch_lammps.sh is properly executed? You can check /src/pair_nequip.cpp and pair_nequip.h

**Comment by Akram-Ibrahim** [comment]
Thanks @JunsuAndrewLee for the update. I will try to use the LAMMPS 23Jun2022 stable version. Yeah, patch_lammps.sh was properly executed and I can see the pair_nequip .cpp and .h files copied in lammps/src.

**Comment by JunsuAndrewLee** [comment]
I also removed 96th line, though not sure this changes anything. Plus, you can easily check available pair styles in your compiled executable with `./lmp -h` command.

**Comment by Linux-cpp-lisp** [comment]
Should be closed by #52.
